nautobot_golden_config/views.py

```diff
@@ -410,7 +410,11 @@ def diff_structured_data(backup_data, intended_data):
 
             settings = get_device_to_settings_map(queryset=Device.objects.filter(pk=device.pk))
             if device.id in settings:
-                _, output = graph_ql_query(request, device, settings[device.id].sot_agg_query.query)
+                sot_agg_query_setting = settings[device.id].sot_agg_query
+                if sot_agg_query_setting is not None:
+                    _, output = graph_ql_query(request, device, sot_agg_query_setting.query)
+                else:
+                    output = None
```

Comment:
I think it should be like this. You have to remove the below now duplicated code, I couldn't get that to be included in the UI.

```suggestion
                sot_agg_query_setting = settings[device.id].sot_agg_query
                if sot_agg_query_setting is not None:
                    _, output = graph_ql_query(request, device, sot_agg_query_setting.query)
                    if structure_format == "yaml":
                        output = yaml.dump(json.loads(json.dumps(output)), default_flow_style=False)
                    else:
                        output = json.dumps(output, indent=4)
                else:
                    output = "No saved `GraphQL Query` query was configured in the `Golden Config Setting`"
```

Reply:
@itdependsnetworks what problem are you running into with the code as it is? I always get `null` (both json and yaml) when there is not a graphql query defined, and the `Detailed View` behaves the same.

Reply:
The introduction of a message means that it would be serialized, and would look odd. Without the message it is not clear to the avg user what is going to happen 

```python
>>> json.dumps(None, indent=4)
'null'
>>> json.dumps('hello', indent=4)
'"hello"'
>>>
```

Reply:
The message in the code I provided

Reply:
alright, I finally understand the original comment. What if we did this?

```
                if sot_agg_query_setting is not None:
                    _, output = graph_ql_query(request, device, sot_agg_query_setting.query)
                else:
                    output = {"Error": "No saved `GraphQL Query` query was configured in the `Golden Config Setting`"}
```

Reply:
works for me :)